server.go

```diff
@@ -318,6 +297,53 @@ func (a *App) reply(w http.ResponseWriter, r *http.Request) {
 	}
 	successReply(w)
 }
+
+func (a *App) run(w http.ResponseWriter, r *http.Request) {
+	var msg Message
+	if err := json.NewDecoder(r.Body).Decode(&msg); err != nil {
+		errorReply(w, http.StatusBadRequest, "couldn't parse json")
+		return
+	}
+
+	msg.TwinSrc = a.twin
+	msg.Retqueue = uuid.New().String()
+
+	// if len(msg.TwinDst) > 1 || msg.TwinDst[0] != a.twin {
+	// 	errorReply(w, http.StatusBadRequest, "couldn't parse json")
+	// 	return
+	// }
+
+	err := a.backend.PushToLocal(r.Context(), msg)
+	fmt.Println(err)
+	if err != nil {
+		errorReply(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	ret := MessageIdentifier{
+		Retqueue: msg.Retqueue,
+	}
+
+	json.NewEncoder(w).Encode(&ret)
+}
+
+func (a *App) getResult(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	var msgIdentifier MessageIdentifier
+	if err := json.NewDecoder(r.Body).Decode(&msgIdentifier); err != nil {
+		errorReply(w, http.StatusBadRequest, "couldn't parse json")
+		return
+	}
+
+	response, err := a.backend.GetMessageReply(r.Context(), msgIdentifier)
```

Comment:
what if an evil user passed `msgbus.system.local` as the return queue? I suggest checking that the queue is a valid uuid and the name of it can act as a token (i.e. its name is random enough that users can't peak on each others queues).

Reply:
resolved :+1: 
 thank you.

![image](https://user-images.githubusercontent.com/36021484/133428886-2c04ec19-21ed-4964-a3e6-4aa6560702b4.png)